util/util.go

```diff
@@ -335,6 +341,54 @@ func GetOwnerMachine(ctx context.Context, c client.Client, obj metav1.ObjectMeta
 	return nil, nil
 }
 
+// GetMachineOwner returns the Unstructured object owning the current resource.
+func GetMachineOwner(ctx context.Context, c client.Client, obj metav1.Object) (*MachineOwner, error) {
+	allowedGKs := []schema.GroupKind{
+		{
+			Group: clusterv1.GroupVersion.Group,
```

Comment:
/hold

I need to fix this to use the controlplane Group. Unfortunately, there's an import cycle between controlplane/api and util. I will fix the cycle in a separate PR to unblock this.

Reply:
I'm not sure there is a need to enforce certain allowed GVKs here, any controller reference should be sufficient for our purpose, since it would indicate ownership of the lifecycle of the resource (that it wasn't created by a user).

I do wonder if instead of just adding the Kind as the value of the annotation if it would make more sense to use GVK instead to avoid any potential for name conflicts/confusion.

Reply:
removed this func altogether

Reply:
/hold cancel